examples/heartbleed/legacy/heartbleed.c

```diff
@@ -0,0 +1,338 @@
+	             5,
+	             5,
+	             LcdFontM5x7_16pt,
+	             "Move Joystick to Change Length.",
+	             BACKGROUND_COLOR,
+	             FOREGROUND_COLOR);
+	lcd_draw_str(lcd,
+	             5,
+	             15,
+	             LcdFontM5x7_16pt,
+	             "Press Joystick to Send.",
+	             BACKGROUND_COLOR,
+	             FOREGROUND_COLOR);
+	lcd_draw_str(lcd,
+	             5,
+	             30,
+	             LcdFontM5x7_16pt,
+	             "Request a larger buffer",
+	             BACKGROUND_COLOR,
+	             FOREGROUND_COLOR);
+	lcd_draw_str(lcd,
+	             5,
+	             40,
+	             LcdFontM5x7_16pt,
+	             "Suggested Length: ",
+	             BACKGROUND_COLOR,
+	             FOREGROUND_COLOR);
+}
+
+/**
+ * @brief Display the request length integer to the LCD.
+ *
+ * @param lcd A handle to Sonata's LCD
+ * @param request_length The request length to write
+ */
+void draw_request_length(St7735Context *lcd, size_t request_length)
+{
+	const size_t ReqLenStrLen = 15u;
+	char         req_len_s[ReqLenStrLen];
+	size_t_to_str_base10(req_len_s, request_length);
+	lcd_draw_str(lcd,
+	             110,
+	             40,
+	             LcdFontM5x7_16pt,
+	             req_len_s,
+	             BACKGROUND_COLOR,
+	             FOREGROUND_COLOR);
+}
+
+/**
+ * @brief This function contains the control logic loop to fetch joystick
+ * input related to the request length until the user presses the joystick
+ * to submit the request.
+ *
+ * @param lcd A handle to Sonata's LCD
+ * @param request_length An out parameter containing the initial request
+ * length, and to which the final request length will be written.
+ */
+void get_request_length(St7735Context *lcd, size_t *request_length)
+{
+	// Initial draw
+	draw_request_length(lcd, *request_length);
+
+	// Get user input
+	DEBUG_LOG("Waiting for user input on the joystick...\n");
+	bool inputSubmitted = false;
+	while (!inputSubmitted)
+	{
+		size_t prev_length = *request_length;
+		inputSubmitted     = length_joystick_control(request_length);
+		if (*request_length == prev_length)
+		{
+			continue; // Only re-draw when changed
+		}
+		lcd_draw_str(lcd,
+		             110,
+		             40,
+		             LcdFontM5x7_16pt,
+		             "       ",
+		             BACKGROUND_COLOR,
+		             FOREGROUND_COLOR);
+		draw_request_length(lcd, *request_length);
+	}
+
+	DEBUG_LOG("Heartbeat submitted with length %d\n", (int)(*request_length));
+}
+
+/**
+ * @brief This function mocks the network and show the package on the lcd
+ * instead.
+ *
+ * @param lcd A handle to Sonata's LCD
+ * @param package The package to be sent.
+ * @param len The length of the package .
+ */
+void network_send(void *handle, const char *package, size_t len)
+{
+	const uint32_t CharsPerLine = 29u;
+	St7735Context *lcd          = (St7735Context *)handle;
+
+	const uint32_t TextAreaBgColor = 0xAAAAAA;
+	const uint32_t TextAreaFgColor = 0x00;
```

Comment:
Nit: maybe add `BGRColorGrey` to `legacy/lcd.h` and instead use:

```suggestion
	const uint32_t TextAreaBgColor = BGRColorGrey;
	const uint32_t TextAreaFgColor = BGRColorBlack;
```